### PR TITLE
Allow required headers to be snake_case in the contacts csv

### LINKED ISSUE #1272: Feature Request: allow csv required headers to be first_name, etc instead of firstName...
Many csv generators from CRMs, etc create headers list `first_name`, `last_name` and some do not support uppercase characters at all.  In order to avoid the need to manually edit csv headers before upload.

**Describe the solution you'd like**
Instead of blocking CSV uploads that have first_name, last_name instead of firstName, etc columns, either header-style should succeed in uploads.

This is entirely on the front-end -- the related code is here:
https://github.com/MoveOnOrg/Spoke/blob/main/src/lib/index.js#L127

Code will probably need to change the validation part a bit and then the function at the bottom `convertRowToContact`
And then, of course, we'll need to change the documentation in src/components/CampaignContactsForm.jsx

spoke_uploadcontacts

### PR DESCRIPTION
**Pull request by lperson**
# Fixes #1272 

## Description

Accept and correctly handle contacts csv files with `first_name` instead of `firstName` and `last_name` instead of `lastName` for their respective column headers.

Updated instructions in contact upload:

Screen Shot 2019-12-21 at 2 17 23 PM

What changed: upgraded papa-parse to version 5 and provided a `transformHeaders` method to implement this.

# Checklist:

- [x] I have manually tested my changes on desktop and mobile
- [x] The test suite passes locally with my changes
- [x] If my change is a UI change, I have attached a screenshot to the description section of this pull request
- [x] [My change is 300 lines of code or less](https://github.com/MoveOnOrg/Spoke/blob/main/CONTRIBUTING.md#your-first-code-contribution), or has a documented reason in the description why it’s longer
- N/A ~[ ] I have made any necessary changes to the documentation~
- [x] I have added tests that prove my fix is effective or that my feature works
- N/A ~[ ] My PR is labeled [WIP] if it is in progress~

### CODE REVIEW

File: src/lib/parse_csv.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    return camelizedColumnHeader;
+  }
+
+  return columnHeader;
+};
+
+export const parseCSV = (file, optOuts, callback) => {
+  Papa.parse(file, {
+    header: true,
+    transformHeader: ensureCamelCaseRequiredHeaders,
```
Comment (schuyler1d) [inline src/lib/parse_csv.js:92]: INFO (no critique): This `transformHeader` option and the ensureCamelCaseRequiredHeaders seems to be the new code, aside from the code movement into a separate file.

File: src/lib/parse_csv.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      optOutCount: optOutRows.length,
+      invalidCellCount: invalidCellRows.length,
+      missingCellCount: missingCellRows.length,
+      zipCount
+    }
+  };
+};
+
+const ensureCamelCaseRequiredHeaders = columnHeader => {
+  const camelizedColumnHeader = humps.camelize(columnHeader);
```
Comment (schuyler1d) [inline src/lib/parse_csv.js:77]: INFO (no critique): humps is also already used in e.g. src/server/api/lib/utils.js so is already a dependency, though this probably adds the library to the frontend now.

File: src/lib/parse_csv.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+  if (
```
Comment (schuyler1d) [inline src/lib/parse_csv.js:79]: Can you add a comment here that just says something like "This preserves the value of `vendor_id` and custom fields/columns"

**Review by schuyler1d — DISMISSED** [review]
:+1: 
This is great!  Made one request to add a comment, but either way, let's merge.

**Review by filafb — DISMISSED** [review]

File: src/lib/parse_csv.js (hunk 3 above)
Comment (lperson) [reply src/lib/parse_csv.js:79]: @schuyler1d I added the comment.

File: src/lib/parse_csv.js (hunk 2 above)
Comment (lperson) [reply src/lib/parse_csv.js:77]: Noted!  Thanks!

**Review by schuyler1d — APPROVED** [review]
:+1:

**Pull request merged**